pkg/kubeadm/cmd/manual.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/renstrom/dedent"
+	"github.com/spf13/cobra"
+
+	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
+	kubemaster "k8s.io/kubernetes/pkg/kubeadm/master"
+	kubenode "k8s.io/kubernetes/pkg/kubeadm/node"
+	kubeadmutil "k8s.io/kubernetes/pkg/kubeadm/util"
+	netutil "k8s.io/kubernetes/pkg/util/net"
+	// TODO: cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+var (
+	manual_done_msgf = dedent.Dedent(`
+		Master initialization complete:
```

Comment:
Yes, I also think that JWS server can still run, even if manual method is
prefer on some nodes... Where is it best to discus this an other big
questions like the default CIDRs we touched above? May be the proposal PR
or a clean issue to address things more specifically?

On Wed, 31 Aug 2016, 22:12 Joe Beda, <notifications@github.com> wrote:

> In pkg/kubeadm/cmd/manual.go
> <https://github.com/kubernetes/kubernetes/pull/31221#discussion_r77074111>
> :
>
> > +	"io"
> > +
> > +	"github.com/renstrom/dedent"
> > +	"github.com/spf13/cobra"
> > +
> > +	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api
>
> "
> > +	kubemaster "k8s.io/kubernetes/pkg/kubeadm/master"
> > +	kubenode "k8s.io/kubernetes/pkg/kubeadm/node"
> > +	kubeadmutil "k8s.io/kubernetes/pkg/kubeadm/util"
> > +	netutil "k8s.io/kubernetes/pkg/util/net"
> > +	// TODO: cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
> > +)
> > +
> > +var (
> > +	manual_done_msgf = dedent.Dedent(`
> > +		Master initialization complete:
>
> Does this need to be a separate command? Or is it really just more of a
> --manual-tls-distribution option for kubeadm init? There are going to be
> other "manual" knobs that we'll want to expose also. For example, you may
> already have a CA that you trust and want to provide certs for the master
> vs. generate them.
>
> It would be worthwhile to noodle on this part of the UX outside of a PR.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31221/files/57096ca902c5c666d45df191037d9278972af1fd#r77074111>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPWS2S8fdcMSOEJG9opu_SnZOT9JUz2ks5qle2ugaJpZM4JqsTH>
> .
>